src/libcrun/linux.c

```diff
@@ -597,6 +672,20 @@ struct device_s needed_devs[] =
     {NULL, '\0', 0, 0, 0}
   };
 
+static const char *
+relative_path_under_dev (const char *path)
+{
+  if (path[0] != '/')
+    return path;
+  if (strncmp (path, "/dev/", 5) == 0)
+    {
+      if (strchr (path + 5, '/'))
```

Comment:
This says to return null if there is a directory in the path?  Is that the intension.

/dev/mapper/foobar
Would return null?
/dev/1234
Would return 1234?

Reply:
yes, correct.  If a path is directly under `/dev`, I can use the faster `mknodat` version and there is no risk of race conditions while it is created, from the `mknodat` man page
```
If pathname already exists, or is a symbolic link, this call fails with an EEXIST error
```

Otherwise if it is not a direct child, a path like `/dev/mapper/foobar` needs to be resolved with the slower `openat` + `check_fd_under_rootfs` variant

Reply:
can you add a comment explaining the logic reasoning then?

Reply:
> can you add a comment explaining the logic reasoning then?

sure, I've pushed a new version that adds some comments